qp_deblur/deblur.py

```diff
@@ -415,19 +420,84 @@ def deblur(qclient, job_id, parameters, out_dir):
             f.write("")
 
     # Step 4, communicate with archive to check and generate placements
-    qclient.update_job_step(job_id, "Step 4 of 4 (1/2): Retriving "
+    qclient.update_job_step(job_id, "Step 4 of 4 (1/4): Retriving "
                             "observations information")
     features = list(load_table(final_biom_hit).ids(axis='observation'))
+
+    fp_phylogeny = None
     if features:
         observations = qclient.post(
             "/qiita_db/archive/observations/", data={'job_id': job_id,
                                                      'features': features})
-        no_placements = [k for k, v in observations.items() if v == '']
-        qclient.update_job_step(job_id, "Step 4 of 4 (2/2): Generating %d new "
-                                "placements" % len(no_placements))
+        novel_fragments = list(set(features) - set(observations.keys()))
+
+        qclient.update_job_step(job_id, "Step 4 of 4 (2/4): Generating %d new "
+                                "placements" % len(novel_fragments))
         try:
+            # Once we support alternative reference phylogenies for SEPP in the
```

Comment:
This is a pretty large try block, can we shorten? The issue with long blocks like this is that we are not sure why/where they raise the issue.